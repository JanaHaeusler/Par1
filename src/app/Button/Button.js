import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const ButtonStyled = styled.button`
  padding: 7px;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 80px;
  height: 34px;
  border-style: none;
  border-radius: 5px;
  box-shadow: 2px 2px 8px var(--secondary-medium-transparent);
  background-color: ${(props) =>
    props.main ? 'var(--primary)' : 'var(--light)'};
  color: ${(props) => (props.main ? 'var(--light)' : 'var(--primary)')};
  font-size: 1rem;

  &:disabled {
    opacity: 50%;
  }
`
const Button = ({ iconComponent, text, ...props }) => (
  <ButtonStyled {...props}>
    {iconComponent}
    {text}
  </ButtonStyled>
)

Button.propTypes = {
  main: PropTypes.bool,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  iconComponent: PropTypes.element.isRequired,
  text: PropTypes.string.isRequired,
}

export default Button
