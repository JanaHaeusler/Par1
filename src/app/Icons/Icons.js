import styled from 'styled-components/macro'
import { ReactComponent as BackIcon } from './assets/arrow-left.svg'
import { ReactComponent as BinIcon } from './assets/bin.svg'
import { ReactComponent as CheckIcon } from './assets/check.svg'
import { ReactComponent as CrossIcon } from './assets/cross.svg'
import { ReactComponent as DetailsIcon } from './assets/details.svg'
import { ReactComponent as LogoIcon } from './assets/golf-logo.svg'
import { ReactComponent as HomeIcon } from './assets/home.svg'
import { ReactComponent as PencilIcon } from './assets/pencil.svg'
import { ReactComponent as PlusIcon } from './assets/plus.svg'

const BackIconPrimary = styled(BackIcon)`
  stroke: var(--primary);
`
const BinIconPrimary = styled(BinIcon)`
  stroke: var(--primary);
`
const BinIconPrimaryText = styled(BinIcon)`
  margin-right: 3px;
  stroke: var(--primary);
`
const CancelIconLightText = styled(CrossIcon)`
  margin-right: 3px;
  stroke: var(--white);
`
const CancelIconPrimaryText = styled(CrossIcon)`
  margin-right: 3px;
  stroke: var(--primary);
`
const CheckIconLightText = styled(CheckIcon)`
  margin-right: 3px;
  width: 25%;
  stroke: var(--white);
`
const DetailsIconPrimary = styled(DetailsIcon)`
  stroke: var(--primary);
`
const PencilIconPrimary = styled(PencilIcon)`
  stroke: var(--primary);
`

export {
  BackIconPrimary,
  BinIconPrimary,
  BinIconPrimaryText,
  CancelIconLightText,
  CancelIconPrimaryText,
  CheckIconLightText,
  DetailsIconPrimary,
  PencilIconPrimary,
  HomeIcon,
  PlusIcon,
  LogoIcon,
}
