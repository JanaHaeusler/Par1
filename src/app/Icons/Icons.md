Icons

```jsx
<CheckIcon/>
<CrossIcon/>
<BinIcon/>
<PencilIcon/>
<PlusIcon/>
<HomeIcon/>
<LogoIcon/>
```
