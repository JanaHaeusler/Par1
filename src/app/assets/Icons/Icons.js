import styled from 'styled-components/macro'

import { ReactComponent as BinIcon } from './bin.svg'
import { ReactComponent as CheckIcon } from './check.svg'
import { ReactComponent as CrossIcon } from './cross.svg'
import { ReactComponent as HomeIcon } from './home.svg'
import { ReactComponent as LogoIcon } from './golf-logo.svg'
import { ReactComponent as PencilIcon } from './pencil.svg'
import { ReactComponent as PlusIcon } from './plus.svg'

const BinIconDark = styled(BinIcon)`
    margin-right: 3px;
    fill: var(--secondary-dark);
`
const CheckIconDark = styled(CheckIcon)`
    margin-right: 3px;
    width: 24%;
    fill: var(--secondary-dark);
`
const CheckIconLight = styled(CheckIcon)`
    margin-right: 3px;
    width: 24%;
    fill: var(--text-light);
`
const CancelIconDark = styled(CrossIcon)`
    margin-right: 3px;
    fill: var(--secondary-dark);
`
const CancelIconLight = styled(CrossIcon)`
    margin-right: 3px;
    fill: var(--text-light);
`
const PencilIconDark = styled(PencilIcon)`
    margin-right: 3px;
    fill: var(--secondary-dark);
`

export { BinIconDark, CheckIconDark, CheckIconLight, CancelIconDark, CancelIconLight, PencilIconDark, HomeIcon, PlusIcon, LogoIcon }