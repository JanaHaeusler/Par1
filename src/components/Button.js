import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Button.propTypes = {
    children: PropTypes.string.isRequired,
  }

export default function Button({children}) {
    return (
        <ButtonStyled>{children}</ButtonStyled>
    )
}

const ButtonStyled = styled.button`
    padding: 7px;
    border-style: none;
    border-radius: 5px;
    box-shadow: 3px 3px 10px lightgrey;
    background-color: var(--primary);
    font-size: 1rem;
`