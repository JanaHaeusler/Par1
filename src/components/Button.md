Regular Button

```js
<Button>&#10003; Save</Button>
```

Disabled Button

```js
<Button disabled>&#10003; Save</Button>
```
