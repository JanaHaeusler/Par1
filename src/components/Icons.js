import { ReactComponent as Check } from '../assets/icons/check.svg'
import { ReactComponent as Cross } from '../assets/icons/cross.svg'
import { ReactComponent as Bin } from '../assets/icons/bin.svg'
import { ReactComponent as Pencil } from '../assets/icons/pencil.svg'
import { ReactComponent as Plus } from '../assets/icons/plus.svg'
import { ReactComponent as Home } from '../assets/icons/home.svg'
import { ReactComponent as Logo } from '../assets/icons/golf-logo.svg'

export { Check, Cross, Bin, Pencil, Plus, Home, Logo }