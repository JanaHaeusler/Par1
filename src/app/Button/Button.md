Enabled Main Button

```jsx
import { CheckIconLight } from '../Icons/Icons'
;<Button
  main
  onClick={() => {}}
  iconComponent={<CheckIconLight />}
  text="Save"
/>
```

Disabled Main Button

```jsx
import { CheckIconLight } from '../Icons/Icons'
;<Button
  main
  onClick={() => {}}
  disabled
  iconComponent={<CheckIconLight />}
  text="Save"
/>
```

Enabled Button

```jsx
import { CheckIconDark } from '../Icons/Icons'
;<Button onClick={() => {}} iconComponent={<CheckIconDark />} text="Save" />
```
