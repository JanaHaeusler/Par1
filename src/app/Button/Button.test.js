import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import { CheckIconWhite } from '../assets/Icons/Icons'
import Button from './Button'

const testProps = {
  main: true,
  disabled: false,
  onClick: jest.fn(),
  iconComponent: CheckIconWhite,
  text: 'Test Text',
}

describe('Button', () => {
  it('shows the right text', () => {
    const { getByText } = render(<Button {...testProps} />)
    expect(getByText('Test Text')).toBeInTheDocument()
  })

  it('is disabled', () => {
    const props = { ...testProps, disabled: true }
    const { getByRole } = render(<Button {...props} />)
    const button = getByRole('button')
    expect(button).toBeDisabled()
  })

  it('is not main button', () => {
    const props = { ...testProps, main: false }
    const { getByRole } = render(<Button {...props} />)
    const button = getByRole('button')
    expect(button).not.toHaveAttribute('main')
  })

  it('calls onClickMock on click', () => {
    const onClickMock = jest.fn()
    const props = { ...testProps, onClick: onClickMock }
    const { getByRole } = render(<Button {...props} />)
    const button = getByRole('button')
    user.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })
})
