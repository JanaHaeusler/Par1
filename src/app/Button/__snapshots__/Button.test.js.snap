// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  height: 34px;
  width: 85px;
  padding: 7px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-shadow: var(--shadow-light);
  border-radius: 5px;
  border-style: none;
  background-color: var(--primary);
  color: var(--white);
  font-family: 'Raleway',sans-serif;
  font-size: 1rem;
  font-weight: 500;
}

.c0:disabled {
  opacity: 50%;
}

<div>
  <button
    class="c0"
  >
    {"attrs":[],"componentStyle":{"rules":["margin-right:3px;width:25%;stroke:var(--white);"],"staticRulesId":"","isStatic":false,"componentId":"Icons__CheckIconLightText-sc-7h7nei-5","baseHash":-1343931257},"foldedComponentIds":[],"styledComponentId":"Icons__CheckIconLightText-sc-7h7nei-5","target":{}}
    Test Text
  </button>
</div>
`;
